### PR TITLE
feat(server): implement LMOVE #369

### PR DESCRIPTION
**Pull request by YuxuanChen98**
Signed-off-by: chenyuxuan.allen <dev.allenchen@gmail.com>

feat(server): implement command "LMOVE" (#369)

changes:
* add support for LMOVE
* reconstruct MOVE-like command (RPOPLUSH)

hope this would help

### CODE REVIEW

**Comment by romange** [comment]
Thanks! Seems that your only github activity was to contribute to Dragonfly :)
If it's ok by you please add yourself to the CONTRIBUTORS file as well.

File: src/server/list_family.cc [hunk 1]
```diff
@@ -876,6 +842,79 @@ void ListFamily::PopGeneric(ListDir dir, CmdArgList args, ConnectionContext* cnt
   }
 }
 
+void ListFamily::MoveGeneric(ConnectionContext* cntx, std::string_view src,
+                                         std::string_view dest, ListDir src_dir,
```
Comment (romange) [inline src/server/list_family.cc:846]: indentation

File: src/server/list_family.cc [hunk 2]
```diff
@@ -338,7 +341,7 @@ OpResult<string> OpRPopLPushSingleShard(const OpArgs& op_args, string_view src,
 
 // Read-only peek operation that determines wether the list exists and optionally
 // returns the first from right value without popping it from the list.
```
Comment (romange) [inline src/server/list_family.cc:343]: the comment should be changes (it's not first on the right anymore). also, `wether` is a typo, please fix on the way :)

**Review by romange — COMMENTED** [review]
Looks very good, a few minor comments.

File: src/server/list_family.cc (hunk 1 above)
Comment (YuxuanChen98) [reply src/server/list_family.cc]: done

File: src/server/list_family.cc (hunk 2 above)
Comment (YuxuanChen98) [reply src/server/list_family.cc]: > the comment should be changes (it's not first on the right anymore). also, `wether` is a typo, please fix on the way :)

done

**Comment by YuxuanChen98** [comment]
> Thanks! Seems that your only github activity was to contribute to Dragonfly :) If it's ok by you please add yourself to the CONTRIBUTORS file as well.

Yes, cause I think Dragonfly would be a great project and I'm really excited to be part of this :)

**Review by romange — APPROVED** [review]

**Comment by romange** [comment]
@YuxuanChen98 welcome to the project! Please join our discord https://discord.gg/HsPjXGVH85 and say hello 😄

**Pull request merged**